### ISSUE TITLE
when i use amp on LSTM network,errro

### ISSUE THREAD

**Issue by daixiangzi** [pytorch/pytorch#42229]
pytorch1.6
ubuntu20.0
code:
for i, (inp, labels) in enumerate(tqdm(train_loader)):
        inp = inp.to(device)
  
        with torch.cuda.amp.autocast():
            preds = model(inp)
            batch_size = inp.size(0)
            text, length = converter.encode(labels)
            preds_size = torch.IntTensor([preds.size(0)] * batch_size)
            loss = criterion(preds, text, preds_size, length)
        optimizer.zero_grad()
        scaler.scale(loss).backward()
        scaler.step(optimizer)
        scaler.update()
erro:
CUDNN_STATUS_BAD_PARAM,
but i use normal mode is ok

cc @mcarilli

**Comment by daixiangzi** [comment]
i try to in forard ,modify inputs = inputs.to(torch.float32) before rnn module,it can use

**Comment by mcarilli** [comment]
Closing as duplicate of https://github.com/pytorch/pytorch/issues/36428.

If you are using `nn.LSTM`, this is a known issue, amp doesn't work with cudnn RNNs.  they're more complicated than other ops because autocast needs to flatten weights in a particular format, in addition to casting them to FP16.

I've written a fix locally that enables autocast with cudnn rnns.  The PR should be up in the next few days.

The cell-based RNN API is expected to work with Amp already.  If you observed this error while using `nn.LSTMCell`, please reopen the issue and submit a minimal repro.
